lib/check-each-pcb-trace-non-overlapping/check-each-pcb-trace-non-overlapping.ts

```diff
@@ -107,6 +107,14 @@ export function checkEachPcbTraceNonOverlapping(
     if (segmentA.x1 === segmentA.x2 && segmentA.y1 === segmentA.y2) continue
 
     for (const obj of nearbyObjects) {
+      // ignore obstacles not on the trace's layer (except vias)
+      if (
+        obj.type !== "pcb_via" &&
```

Comment:
i mentioned in my comment, we actually do want to handle vias, vias have a layers property indicating what layers they're on, so as long as they return that from `getLayersOfPcbElement` it should work yea?